site/src/pages/api/rewrites/sandboxed-block-demo.api.ts

```diff
@@ -0,0 +1,133 @@
+const hotfixExternals = (externals: JSONObject | undefined): JSONObject => {
+  return (
+    externals ?? {
+      react: "17.0.2",
+      lodash: "4.17.21",
+      twind: "0.16.16",
+    }
+  );
+};
+
+/**
+ * @todo potentially remove after building blocks to ESM
+ */
+const hotfixPackageName = (packageName: string): string => {
+  return packageName === "lodash" ? "lodash-es" : packageName;
+};
+
+const handler: NextApiHandler = async (req, res) => {
+  if (req.url?.startsWith("/api/rewrites/")) {
+    res.status(404).send("Not found");
+    return;
+  }
+
+  const catalog = readBlocksFromDisk();
+
+  const packagePath = `${req.query.shortname}/${req.query.blockslug}`;
+
+  const blockMetadata = catalog.find(
+    (metadata) => metadata.packagePath === packagePath,
+  );
+
+  if (!blockMetadata) {
+    res.status(404);
+    res.write("Block not found");
+    res.end();
+    return;
+  }
+
+  const { source } = await readBlockDataFromDisk(blockMetadata);
+  const reactVersion = blockMetadata.externals?.react ?? "17.0.2";
+  const mockBlockDockVersion = "0.0.9";
+
+  const externalUrlLookup: Record<string, string> = {};
+
+  const externals = hotfixExternals(blockMetadata.externals);
+
+  for (const [packageName, packageVersion] of Object.entries(externals)) {
+    externalUrlLookup[packageName] = `https://esm.sh/${hotfixPackageName(
+      packageName,
+    )}@${packageVersion}`;
+  }
+
+  const html = `
+    <script type="module">
+      const handleMessage = ({ data }) => {
+        if (typeof data !== "string") {
+          return;
+        }
+        globalThis.initialBlockProps = JSON.parse(data);
+        window.addEventListener("message", handleMessage);
+      }
+      window.addEventListener("message", handleMessage, false);
+    </script>
+    <script type="module">
+      import React from "https://esm.sh/react@${reactVersion}"
+      import ReactDOM from "https://esm.sh/react-dom@${reactVersion}"
+      import { jsx as _jsx } from "https://esm.sh/react@${reactVersion}/jsx-runtime.js";
+      import { MockBlockDock } from "https://esm.sh/mock-block-dock@${mockBlockDockVersion}?alias=lodash:lodash-es&deps=react@${reactVersion}"
+
+      const requireLookup = {
+        "react-dom": ReactDOM,
+        react: React,
+        ${Object.entries(externalUrlLookup)
+          .map(
+            ([packageName, packageUrl]) =>
+              `"${packageName}": await import("${packageUrl}")`,
+          )
+          .join(",")}
+      }
+
+      const require = (packageName) => {
+        return requireLookup[packageName];
+      };
+
+      const loadCjsFromSource = (source) => {
+        const module = { exports: {} };
+        const moduleFactory = new Function("require", "module", "exports", source);
+        moduleFactory(require, module, module.exports);
+
+        return module.exports;
+      }
+
+      const findComponentExport = (module) => {
+        return module.default ?? module.App ?? module[Object.keys(module)[0]];
+      }
```

Comment:
Can we throw if this isn't found?

Reply:
https://github.com/blockprotocol/blockprotocol/pull/312/commits/a1db5a63fcf9e8a186ac790955b28c4e1ed1d401